ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java

```diff
@@ -476,7 +477,7 @@ public void getTableList(Context ctx) {
           .filter(e -> !arcticTables.contains(e))
           .forEach(e -> tables.add(new TableMeta(e, TableMeta.TableType.HIVE.toString())));
     }
-
+    tables.sort((Comparator.comparing(TableMeta::getName)));
```

Comment:
Sorting here will mix the result of arctic tables and hive tables, how about sorting them separately?

Reply:
Well, the final result returned is the mix of arctic tables and hive tables, I have no idea about the meaning of sorting separately

Reply:
`sorting separately` means sorting arctic tables and hive tables separately, so all arctic tables will be placed before all hive tables.

By sorting separately rather than having them mixed together, users can quickly identify the existence of Arctic/Hive tables under a database.

Reply:
@wangtaohz Are you suggesting that we should sort the returned results in the order of (format, table_name) instead of just (table_name)? Considering that the Catalog may indeed return tables with multiple formats, I'm not sure if we need to aggregate tables of different formats together. Additionally, do tables of different formats have any priority or precedence?

Reply:
Yes, I believe it is better **NOT** to mix tables with different formats together, and sorting the results in the order of (format, table_name) is a good implementation method. 

As for the priority of different formats,  maybe sorting them in alphabetical order is acceptable, which is `arctic`,`hive`,`iceberg`,`paimon`.

The code may be like
```java
    tables.sort(
        (table1, table2) -> {
          if (Objects.equals(table1.getType(), table2.getType())) {
            return table1.getName().compareTo(table2.getName());
          } else {
            return table1.getType().compareTo(table2.getType());
          }
        });
```

Reply:
However, Hive format tables are quite different here in that they are not managed by AMS. AMS only displays them for the purpose of upgrades. Their priority seems to be lower.

Besides, the `Arctic` table type is out-of-date, we may use the format here directly, but it may require frontend changes, we can optimize it by other PR.

Reply:
Perhaps we should go back to the purpose of this pr, as described in the issue #1716 . The main purpose is to quickly locate a table among multiple tables, so it is more effective to do global sorting tables returned by a catalog. 


Considering that front-end displays now don't distinguish between formats, I think it is not very helpful for solving the problem of "quickly locate a table"  if the back-end separate sorting. And  separate sorting will bring more difficult to locate a table, and the code readability deteriorates, and you need to explain why this should be sorted separately.

Reply:
Two aspects:
First, for this PR corresponding to the Issue, the purpose is to conveniently and quickly find the tables of interest. If I aim to look for tables in Iceberg format that start with the letter "b," but the sorting is based on (format, table_name), I might encounter a Paimon format table or a Mix format table first, which is not convenient.

Second, considering the current catalog architecture, we have different formats under one catalog. Mixing these formats together is neither aesthetically pleasing nor convenient.

Therefore, I believe we indeed need to sort according to (format, table_name), but the frontend also needs to change. The frontend needs to set up dropdowns (select boxes or others) for different formats under the catalog to display tables of different formats. This approach is both aesthetically pleasing and convenient, addressing various needs. For example, if I want to find an Iceberg table starting with the letter "b," I only need to click on "Iceberg format" and skip past tables starting with the letter "a" to find those starting with "b," without having to skip over tables of other formats that might exist above.
Something like this:
![image](https://github.com/NetEase/amoro/assets/68625618/32c2ebcd-0411-446f-af7e-abf143a90dd5)